### DISCUSSION TITLE
Add to Basket AJAX Button Issue on Development Environment After Migration to Bootscore 6.0.2 - Request for comment

### DISCUSSION THREAD

**Discussion by hsankala** [bootscore/bootscore#879] [Q&A] [answered]
I’m currently using Bootscore 5.3.4 on a customer’s website, [Church Antiques](https://churchantiques.com/), and I’m in the process of migrating to version 6.0.2. I’ve been checking the [migration guide provided](https://bootscore.me/documentation/migration/) on the Bootscore website and installed the latest version of Bootscore parent theme (6.0.2) on a development environment on my local PC. My plan is to use a child theme, but before proceeding, I wanted to do a quick sanity check to ensure that the base theme (6.0.2) works fine on its own.

After testing, I encountered an issue on the product pages: the “Add to Basket” button triggers the AJAX animation for adding to the cart, but it never completes, and the cart page does not open. I have a short video demonstrating this behavior:  https://youtu.be/IJ6lhev7oK8

Here are the details of the development website migration:

Version: Bootscore 6.0.2  (Both non-modified parent and child theme's exhibit this behavour)
Plugin: The latest version of WooCommerce (v9.4.1)
Plugins: Only WooCommerce is enabled on the site.

Setup: This is on the customer’s live site data has been used on this development site migrating from version 5.3.4. Database copied over and filesystem.

Development Test: I tried running Bootscore 6.0.2 on a clean WordPress install in my local development environment, and the issue does not appear there.

I haven’t seen any errors in the browser console on the products page when performing this action or in the WP debug file. However, I’m still unsure about the root cause. The problem is only present on the customer’s site that I've copied over to my development environment with the default Bootscore theme, without any customizations or child themes applied. The only issue I’ve noticed so far is with the AJAX cart on the product page, everything else is working fine!

Could you kindly suggest where I should look to debug this issue or what might be causing it? Any guidance on potential areas to check would be greatly appreciated.

**Comment by hsankala** [comment]
Oooo I think I’ve figured out what might be causing the issue with the AJAX "Add to Basket" functionality. When I went to the WooCommerce Settings > Products tab, I noticed the option for Redirect to the basket page after a successful addition was checked. 

![image](https://github.com/user-attachments/assets/28d97983-494f-4cd8-ba49-1aac6eb1f66a)

This was an option that got enabled when I restored the site to my local development environment, but this checkbox is not selected on the live production environment.

After deselecting the Redirect to the basket page after successful addition option, the AJAX add to basket functionality on the product page now works as expected. The cart sidebar is updated and slides out without needing a page refresh.

I’m wondering why this redirect option would cause a conflict with the AJAX add-to-cart behavior. My assumption is that when the option is enabled, it forces a redirect to the cart page, which might be interfering with the expected behavior on the product page.

Could this be considered a bug, or is there something in WooCommerce’s handling of this option that could be causing this behavior?

I’ve tested this behavior on a bare-bones, new installation of WordPress (latest version), with the latest WooCommerce version, bootscore (latest) and using the standard WooCommerce sample data import. When the 'Redirect to the basket page after a successful addition' option is selected, I get the same behavior: on the product page, the "Add to Cart/Add to Basket" button only shows the animation, and the cart slider doesn’t appear.

![image](https://github.com/user-attachments/assets/d985438d-7fff-4f2a-9c2f-33ea86aa99b7)}

Screen recording of this issue here: https://youtu.be/K2ZvSNkNAfo

**Reply by crftwrk** [reply, marked as answer]
Yes, that's because since v6 there is no cart page anymore. The offcanvas mini-cart is now a full-featured cart and there is no need for an additional cart page. Offcanvas cart has no View cart button, only a Checkout button to create a smoother shopping experience. 

If you want to use "Redirect to the cart page after successful addition", then you need two filters. One to enable the cart page and one that disables the AJAX in product pages:

```php
/**
 * Enable cart page
 */
add_filter('bootscore/skip_cart', '__return_false');


/**
 * Disable AJAX cart
 */ 
add_filter('bootscore/load_ajax_cart', '__return_false');
```
- View cart button in mini-cart is enabled (add something to the cart to trigger AJAX)
- Cart page is enabled
- Quantity buttons in mini-cart are removed

See https://bootscore.me/documentation/woocommerce/#restoring-woocommerce-s-default-functionality

**Reply by hsankala** [reply]
Awesome thanks makes total sense now!!

**Reply by crftwrk** [reply]
You're welcome 👍
